Update zwave_js documentation for support (#20378)
home-assistant/home-assistant.github.io: source/_integrations/zwave_js.markdown

  - switch
---

> This integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver. This is our recommended Z-Wave integration for Home Assistant. If you're new to Home Assistant, use Z-Wave JS. The `zwave` and `ozw` integrations have been marked as deprecated and will no longer receive any updates. See [our migration instructions](#migrating-from-previous-z-wave-implementations) to migrate from a prior integration.

Reviewer: This page is specifically towards Z-Wave JS, do we need to mention zwave/ozw at this point even?

The `zwave` and `ozw` pages have large alerts on their deprecation already.

Author: I just relocated that portion I think. I defer to whomever. We need a specific reference to the migration instructions at the top. That's all I care about.

Reviewer: I agree, however, it isn't written from a migration focus, but from a "new Home Assistant" perspective. The line even starts with: "If you're new to Home Assistant", which is not the case for user migrating from `zwave` or `ozw`.

Reviewer: Unresolving this one, as per my last comment.

I think we should rewrite this as something like:

"If you are currently using one of the deprecated `zwave` or `ozw` integrations, see [our migration instructions](#migrating-from-previous-z-wave-implementations) to migrate from a prior integration."

Author: That's nearly what I wrote verbatim? You're looking at the old one.

Reviewer: I definitely looked at the old version indeed 👍 

Let's still remove "If you're new to Home Assistant, use Z-Wave JS.". This piece of text on its own makes no sense in that context IMHO.

To Run Z-Wave JS you will need a [Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules), a running Z-Wave JS server (using only **one** of the add-ons or installation methods described below), and the Z-Wave JS integration.

{% include integrations/config_flow.md %}

If you run Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Z-Wave JS Server add-on, please see the [advanced installation instructions](#advanced-installation-instructions).

> For new installations, network security keys will be automatically generated for you. If this Z-Wave stick has already paired with secure devices, you need to enter the previously used network key as the S0 network key. S2 security keys will be automatically generated for you. Make sure that you keep a backup of these keys in a safe place in case you need to move your Z-Wave stick to another device.

Reviewer: ```suggestion
For new installations, network security keys will be automatically generated for you. If this Z-Wave stick has already been paired with secure devices, you need to enter the previously used network key as the S0 network key. S2 security keys will be automatically generated for you. Make sure that you keep a backup of these keys in a safe place in case you need to move your Z-Wave stick to another device.
```

This integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver. This is our recommended Z-Wave integration for Home Assistant.

> To Run Z-Wave JS you will need a [Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules), a running Z-Wave JS server (using only **one** of the addons or installation methods described below), and the Z-Wave JS integration.

Reviewer: ```suggestion
To Run Z-Wave JS you will need a [Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules), a running Z-Wave JS server (using only **one** of the add-ons or installation methods described below), and the Z-Wave JS integration.
```

Additional devices may be discoverable, however only devices that have been confirmed discoverable are listed above.

**700 series USB sticks are fully supported by Z-Wave JS and are recommended for new users purchasing their first USB stick**, but those USB sticks are not presently auto-discoverable by the integration.

> ## Migrating from previous Z-Wave implementations

Reviewer 2: I think we should move the migration and advanced installation sections back down. It's not something that users will need to use more than once, for the most part.

Author: I debated that, but users don't notice them at the bottom as it's contrary to how documentation is usually organized. Is it possible to have two pages for the integration and link to the installation and migration page at the top? I'm not sure that helps with the noticing them but at least it would allow better organization and better headings.

Reviewer 2: We already have a link to the advanced instructions in the quickstart section at the top. We could add a link to migration there. Something like:

```
If you're migrating from one of the deprecated Z-Wave integrations, `zwave` or `ozw`, please see the [migration section](#migrating-from-previous-z-wave-implementations).
```

Author: I know that we do, I'm just saying that I've seen dozens of people say they looked at that and they didn't know the instructions were there, or they didn't see part of them. So what's there isn't terribly effective... I think it's actually made worse that the quick start instructions are at the top and then they skip the link. 

What if we move all the installation instructions down and have a link to those at the top? That way they aren't enticed to just look at the ones that are there?

Reviewer 2: We want to promote the simple installation on HA OS with official add-on. It's ok if the advanced instructions is a bit less prominent.

If I google for "Home Assistant zwave js migration" I can link directly into the migration section from the top google result. It doesn't seem hard to find for me.

Author: If you want I'll stet back the move and suggest other changes to the quick start section. That section isn't complete. It doesn't make clear they need one addon and the integration, as an example, which is a common question. Nor does it mention the migration wizard leading to frequent questions whether one exists (that's not really an advanced installation in any event). 

Nor are the requirements clear at the top of the integration page...that's not really part of the advanced installation instructions either. 

I'd expect the documentation to be:
- Requirements
- Installation
- Use

If we're not going to follow that standard format then I think we need a roadmap at the top pointing to those sections. It's just not very user friendly right now, especially for less advanced users. It essentially assumes someone is familiar with what's necessary to run zwave and (most likely) the old integration...

There is also repetition right now. E.g. Configuration (with a drop down for manual) duplicates the section titled "Installing and Configuring the Zwave JS Integration."

I'm happy to help clean this up, but my recommendation is to be less wedded to what we have.

Reviewer 2: The goal with the integration is to allow users running HA OS a plug and play experience where they can plug in the USB stick, go to the integrations page, and easily set up the integration with a couple of clicks. This is what the quick start section reflects. The user shouldn't need to know more at that point. It also mentions the advanced instructions if the user isn't running HA OS or wants to read and learn more.

Author: I understand the goal and intent. Part of an easy user experience is being able to find an answer when you have a question, though. While the intent is for it to be one click lots of people know and have heard there is more than one option. They then have difficulty understanding what's required. Even without zwavejs2mqtt users are confused and frequently install the addon and no integration. I'm just saying let's add a short section at the top that says:

- If you want to run zwavejs, you need a usb stick (link to requirements) and to install the integration (link to configuration section), which will then install the zwavejs addon automatically. 

- To migrate from the prior zwave integration, use the migration wizard (link to that section).

- To instead use the zwavejs2mqtt addon with the integration, see the advanced installation instructions (link to advanced instructions).

Then remove the duplication. If configuration is going to be at the top, then the advanced instructions should link to there and say "Continue with configuration".

By that regard, the quick start should really just be the Configuration section anyway. They click the link and it installs. Instead that process is covered three times in three different places (quick start, Configuration, and under Advanced). But the quick start doesn't even have the link to install it, that's under Configuration. 

Discovery via USB is a cool feature but is even less relevant to the user documentation wise. Either it works and this is all done for them or it doesn't. If that's covered in the user guide at all it should be an FAQ question.

Author: In fact, what if we cut everything above Configuration and add the links to migration and advanced under that. Then get rid of Installing and Configuring under advanced. The point about waking up devices and interviewing should be a note under Configuration as that has nothing to do with "using zwave". And move Discovery via USB to an FAQ question? That would go a long way...

I'd also divide up the FAQ into sections on Installation, Configuration, Use, and Troubleshooting with the questions under those headers, which will clean up the current "On This Page" section so it's more usable.

Reviewer 2: Your last suggestion sounds good!

Author: Ok. I'll make those changes later and push them.

Author: How about these changes?

| `trigger.current_value`      | The current value for this Z-Wave value (translated to a state name when possible).        |
| `trigger.current_value_raw`  | The raw current value for this Z-Wave value (the key of the state when a state is named).  |

> ## Migrating from previous integrations

Reviewer 2: Let's not change this header. It's an unnecessary change of a link.

> It is easy to switch over from `zwave` or `ozw` integrations to the new Z-Wave JS integration. Your entire Z-Wave network is **stored on your stick** so you will not have to recreate your network. A few things, such as how you receive [central scene events](#events) will work differently than they did before.

Reviewer: Terms like "easy" we should avoid. These are subjective and can be received condescending for people that don't experience it as such.

### Automatic migration wizard

> If you are using the `zwave` integration, there is a migration wizard that will help you setup the Z-Wave JS integration, remove the `zwave` integration and migrate the entities and devices that can be mapped from the `zwave` integration to the Z-Wave JS integration. The migration wizard is available from the `zwave` integration configuration panel in the GUI.

Reviewer: The "setup" changing is incorrect in this context. It's not about "his setup" but about "helping setting up". Two words.

```suggestion
If you are using the `zwave` integration, there is a migration wizard that will help you set up the Z-Wave JS integration, remove the `zwave` integration and migrate the entities and devices that can be mapped from the `zwave` integration to the Z-Wave JS integration. The migration wizard is available from the `zwave` integration configuration panel in the GUI.
```

Some entities may not be able to migrate automatically and you will need to rename the corresponding available Z-Wave JS entities manually. Before completing the migration you will be shown a list of entities that could not be migrated automatically, and you'll have the option to abort or continue with the migration.

> There is no automatic migration wizard for the `ozw` integration. Please follow the manual migration path below if you want to migrate from `ozw` to Z-Wave JS.
> ### Manual migration path

Reviewer: ```suggestion
There is no automatic migration wizard for the `ozw` integration. Please follow the manual migration path below if you want to migrate from `ozw` to Z-Wave JS.

### Manual migration path
```

### Running [Z-Wave JS Server](https://github.com/zwave-js/zwave-js-server).

This application provides the connection between your Z-Wave USB stick and Home Assistant. The Home Assistant Z-Wave JS integration connects to this server via a websocket connection. You need to run this Z-Wave server before you can use the integration.

There are multiple ways to run this server:

**Option 1: The official Z-Wave JS add-on, as described above**

_This option is only available for Home Assistant OS (the recommended installation type) and Home Assistant Supervised installations._

This add-on can only be configured via the built-in Z-Wave JS control panel in Home Assistant.

**Option 2: The Zwavejs2Mqtt (aka "Z-Wave JS to MQTT") add-on installed from the community add-on store**

_This option is only available for Home Assistant OS (the recommended installation type) and Home Assistant Supervised installations._

This add-on includes the Z-Wave JS Server as part of the Zwavejs2Mqtt application. Despite the name, MQTT is not required to run this add-on and both provide the same Z-Wave JS driver.

> The Z-Wave network can be configured via the built-in Z-Wave JS control panel in Home Assistant and alternatively via the Z-Wave JS control panel built into Zwavejs2Mqtt. It provides you with a full-fledged, attractive and feature-complete UI to manage your Z-Wave nodes and settings, which may support more advanced use cases as development continues on the Z-Wave JS control panel.

Reviewer: ```suggestion
The Z-Wave network can be configured via the built-in Z-Wave JS control panel in Home Assistant and alternatively via the Z-Wave JS control panel built into Zwavejs2Mqtt. It provides you with a full-fledged, attractive, and feature-complete UI to manage your Z-Wave nodes and settings, which may support more advanced use cases as development continues on the Z-Wave JS control panel.
```

**Option 3: The Zwavejs2Mqtt Docker container**

This is the recommended approach if you're running Home Assistant Container. See the [Zwavejs2Mqtt documentation](https://zwave-js.github.io/Zwavejs2Mqtt/#/getting-started/quick-start) for instructions.

This method provides the same server application and UI as the Zwavejs2Mqtt add-on. After installing the Docker image, make sure you enable the WS Server in the Home Assistant section of Settings page.

**Option 4: Run the Z-Wave server yourself**

This is considered a very advanced use case. In this case you run the Z-Wave JS Server or Zwavejs2Mqtt NodeJS application directly. Installation and maintaining this is out of scope for this document. See the [Z-Wave JS server](https://github.com/zwave-js/zwave-js-server) or [Zwavejs2Mqtt](https://github.com/zwave-js/zwavejs2mqtt) GitHub repository for information.

<div class='note info'>

> Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). The Z-Wave controller dongle should be connected to the same host as where the Z-Wave JS server is running. In the configuration for the Z-Wave server you need to provide the path to this stick. It's recommended to use the `/dev/serial-by-id/yourdevice` version of the path to your stick, to make sure the path doesn't change over reboots. The most common known path is `/dev/serial/by-id/usb-0658_0200-if00`.

Reviewer: ```suggestion
Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). The Z-Wave controller dongle should be connected to the same host as where the Z-Wave JS server is running. In the configuration for the Z-Wave server, you need to provide the path to this stick. It's recommended to use the `/dev/serial-by-id/yourdevice` version of the path to your stick, to make sure the path doesn't change over reboots. The most common known path is `/dev/serial/by-id/usb-0658_0200-if00`.
```

</div>

<div class='note info'>

> **Network keys** are used to connect securely to compatible devices. The network keys consists of 32 hexadecimal characters, for example `2232666D100F795E5BB17F0A1BB7A146` (do not use this one, pick a random one). Without network keys security enabled devices cannot be added securely and will not function correctly. You must provide these network keys in the configuration part of the Z-Wave JS Server.

Reviewer: ```suggestion
**Network keys** are used to connect securely to compatible devices. The network keys consist of 32 hexadecimal characters, for example, `2232666D100F795E5BB17F0A1BB7A146` (do not use this one, pick a random one). Without network keys security enabled devices cannot be added securely and will not function correctly. You must provide these network keys in the configuration part of the Z-Wave JS Server.
```

> For new installations, unique default keys will be auto generated for you by the Z-Wave JS add-on. You can also generate those network keys in Settings section of Zwavejs2Mqtt.

Reviewer: ```suggestion
For new installations, unique default keys will be auto-generated for you by the Z-Wave JS add-on. You can also generate those network keys in the Settings section of Zwavejs2Mqtt.
```

If migrating from the `zwave` or `ozw` integrations, your network key from those integration should be entered as the S0 network key. Those integrations did not support S2 security, so you will not yet have S2 network keys to configure.

Make sure that you keep a backup of these keys in a safe place. You will need to enter the same keys to be able to access securely paired devices.

</div>

### Installing and configuring the Z-Wave JS integration in Home Assistant

Once you have the Z-Wave server up and running, you need to install and configure the integration in Home Assistant (as described above).

> If you're running full Home Assistant with supervisor, you will be presented with a dialog that asks if you want to use the Z-Wave JS Supervisor add-on. You **must** uncheck this box if you are running the Z-Wave JS server in any manner other than the official Z-Wave JS add-on, including using Zwavejs2Mqtt add-on.

Reviewer: ```suggestion
If you're running full Home Assistant with supervisor, you will be presented with a dialog that asks if you want to use the Z-Wave JS Supervisor add-on. You **must** uncheck this box if you are running the Z-Wave JS server in any manner other than the official Z-Wave JS add-on, including using Z-Wave JS to MQTT add-on.
```

> If you're not running the supervisor or you've unchecked the above mentioned box, you will be asked to enter a websocket URL (defaults to ws://localhost:3000). It is very important that you fill in the correct (docker) IP/hostname here. For example for the Zwavejs2Mqtt add-on this is `ws://a0d7b954-zwavejs2mqtt:3000`.

Reviewer: ```suggestion
If you're not running the supervisor or you've unchecked the above mentioned box, you will be asked to enter a websocket URL (defaults to ws://localhost:3000). It is very important that you fill in the correct (docker) IP/hostname here. For example for the Z-Wave JS to MQTT add-on this is `ws://a0d7b954-zwavejs2mqtt:3000`.
```

## Advanced installation instructions

If you are using Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Z-Wave JS Server add-on, you will need to run the Z-Wave JS server yourself, to which the Z-Wave JS integration will connect.

### Running [Z-Wave JS Server](https://github.com/zwave-js/zwave-js-server).

This application provides the connection between your Z-Wave USB stick and Home Assistant. The Home Assistant Z-Wave JS integration connects to this server via a websocket connection. You need to run this Z-Wave server before you can use the integration.

There are multiple ways to run this server:

**Option 1: The official Z-Wave JS add-on, as described above**

_This option is only available for Home Assistant OS (the recommended installation type) and Home Assistant Supervised installations._

> This add-on can only be configured via the built-in Z-Wave JS control panel in Home Assistant. While that control panel gains new features with every release, some advanced configurations still require the Zwavejs2Mqtt UI.

Reviewer 2: We shouldn't go into details what the add-ons or control panels support. That's in constant flux.
```suggestion
This add-on can only be configured via the built-in Z-Wave JS control panel in Home Assistant.
```

Author: This is a very common question, though. People ask all the time what the benefit of the zjs2mqtt is. Or they get upset they have to change because they want to do something advanced and didn't know to investigate if it was supported. It changes but that's why I left it generic. Do you really think we shouldn't acknowledge that at this point there are still some limitations in the built-in UI?

Author: How about this, I dropped that line but changed the zjs2mqtt one to read: "The Z-Wave network can be configured via the built-in Z-Wave JS control panel in Home Assistant and alternatively via the Z-Wave JS control panel built into Zwavejs2Mqtt. It provides you with a full-fledged, attractive and feature-complete UI to manage your Z-Wave nodes and settings, which may support more advanced use cases as development continues on the Z-Wave JS control panel."

Z-Wave JS supports all known 500 and 700 series Z-Wave controllers. If you are just starting out, we recommend that you purchase a 700 series controller to take advantage of the latest improvements.

For more information, see [Supported Z-Wave dongles](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules)

#### Why was I (or why was I not) automatically prompted to install Z-Wave JS?

Some Z-Wave USB sticks can be auto-discovered, which can simplify the Z-Wave JS setup process. The following devices have been tested with discovery, and offer a quick setup experience; however, these are **not** all of the devices supported by Z-Wave JS:

| Device | Identifier | Vendor |
| -------| ---------- | ------ |
| Aeotec Z-Stick Gen5+ | 0658:0200 | https://aeotec.com/z-wave-usb-stick/ |
| Nortek HUSBZB-1 | 10C4:8A2A | https://www.nortekcontrol.com/products/2gig/husbzb-1-gocontrol-quickstick-combo/ |
| Zooz ZST10 | 10C4:EA60 | https://www.getzooz.com/zooz-zst10-s2-stick.html |
| Z-WaveMe UZB | 0658:0200 | https://z-wave.me/products/uzb/ |

Additional devices may be discoverable, however only devices that have been confirmed discoverable are listed above.

> 

Reviewer 2: ```suggestion
```

> 700 series USB sticks are not presently auto-discoverable by the integration.

Reviewer 2: This isn't true. My Aeotec Z-Stick 7 is automatically discovered.
```suggestion
```

Author: Not anymore. It did in the first release but there were too many false positives. I was the one that put @bdraco in touch with the manufacturers to see if they could set identifiers as right now the fingerprint hits on everything using that same serial adapter. Fine if you want to take it out, though.

**Option 3: The Zwavejs2Mqtt Docker container**

This is the recommended approach if you're running Home Assistant Container. See the [Zwavejs2Mqtt documentation](https://zwave-js.github.io/Zwavejs2Mqtt/#/getting-started/quick-start) for instructions.

This method provides the same server application and UI as the Zwavejs2Mqtt add-on. After installing the Docker image, make sure you enable the WS Server in the Home Assistant section of Settings page.

**Option 4: Run the Z-Wave server yourself**

This is considered a very advanced use case. In this case you run the Z-Wave JS Server or Zwavejs2Mqtt NodeJS application directly. Installation and maintaining this is out of scope for this document. See the [Z-Wave JS server](https://github.com/zwave-js/zwave-js-server) or [Zwavejs2Mqtt](https://github.com/zwave-js/zwavejs2mqtt) GitHub repository for information.

<div class='note info'>

> Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). The Z-Wave controller dongle should be connected to the same host as where the Z-Wave JS server is running. In the configuration for the Z-Wave server, you need to provide the path to this stick. It's recommended to use the `/dev/serial-by-id/yourdevice` version of the path to your stick, to make sure the path doesn't change over reboots. The most common known path is `/dev/serial/by-id/usb-0658_0200-if00`.

Reviewer 2: ```suggestion
[Supported Z-Wave dongle](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). The Z-Wave controller dongle should be connected to the same host as where the Z-Wave JS server is running. In the configuration for the Z-Wave server, you need to provide the path to this stick. It's recommended to use the `/dev/serial-by-id/yourdevice` version of the path to your stick, to make sure the path doesn't change over reboots. The most common known path is `/dev/serial/by-id/usb-0658_0200-if00`.
```

{% endraw %}

#### Can I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

You can, but you cannot run them both at the same time. Only one of them can be active at the same time.

#### How do I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

Switching is easy and does not require renaming your devices.

1. Note your network security keys from your current add-on, then uninstall that add-on. **Do not remove the Z-Wave JS integration or you will lose all device and entity naming.**

2. Install and configure the new add-on, including setting those network security keys.

3. Reconfigure the Z-Wave integration in Home Assistant by clicking "Configure" for Z-Wave JS on the Integrations page, then select "Re-Configure Server". Add the the new/updated WebSocket URL.

#### What's the benefit of using Zwavejs2Mqtt over the official Add-On?

> The official add-on provides the Z-Wave Server in it's bare minimum variant, just enough to serve the Home Assistant integration.

Reviewer: ```suggestion
The official add-on provides the Z-Wave Server in its bare minimum variant, just enough to serve the Home Assistant integration.
```

#### Zwavejs2Mqtt seems to provide discovery of Home Assistant devices on its own too, now I'm confused

Correct, the Zwavejs2Mqtt project existed before Home Assistant had plans to move to the Z-Wave JS Driver. You should use the integration for device discovery and _not_ the MQTT discovery provided by Zwavejs2Mqtt.

#### Can I run Zwavejs2Mqtt only for the control panel and nothing else?

Sure, in the settings of Zwavejs2Mqtt, make sure to enable "WS Server" and disable "Gateway".

#### How do I use my OZW network key in Zwavejs2Mqtt?

You can use your existing network key in Zwavejs2Mqtt but you need to slightly adjust it.
The OZW looks like this: `0x01, 0x02, 0x03 etc.` while the network key format accepted in Zwavejs2Mqtt looks like this `0102030405 etc.`. You can simply edit your existing key and remove the `"0x"` part and the `", "` part so it becomes one large string of numbers.

#### Should I name my devices in Home Assistant, or in Zwavejs2Mqtt?

> Ultimately, this is a personal decision. If you provide a name or location for a device in the Zwavejs2Mqtt UI, that name will be imported into Home Assistant when the integration is reloaded or Home Assistant is restarted. Any entity names, however, will not change if the device has already been setup by Home Assistant. Names set in Zwavejs2Mqtt _will not_ overwrite changes that have already been made in Home Assistant.

Reviewer: ```suggestion
Ultimately, this is a personal decision. If you provide a name or location for a device in the Zwavejs2Mqtt UI, that name will be imported into Home Assistant when the integration is reloaded or Home Assistant is restarted. Any entity names, however, will not change if the device has already been set up by Home Assistant. Names set in Zwavejs2Mqtt _will not_ overwrite changes that have already been made in Home Assistant.
```

{% endraw %}

#### Can I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

You can, but you cannot run them both at the same time. Only one of them can be active at the same time.

#### How do I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

Switching does not require renaming your devices.

> 1. Note your network security keys from your current add-on, then uninstall that add-on. **Do not remove the Z-Wave JS integration or you will lose all device and entity naming.**

Reviewer 2: This is not a good advice as the integration will try to install the add-on again when losing the connection. To do it this way you first need to disable the integration, then you can uninstall the add-on. But then we can't reconfigure the server from the configuration panel, if the integration is disabled.

To use the reconfigure feature in the configuration panel, the flow should probably start with installing the new add-on and configuring it, but not starting it. Then go to reconfigure the server in the configuration panel. When in the manual step, start the new add-on, wait until it's up, then enter the address in the manual step and finish the flow.

This will not currently work though since we need to fix the re-config flow to disconnect and stop the official add-on first when the manual step is selected.

For now, the working flow is to:
1. Disable the integration, this will also stop the official add-on.
2. Install, configure and start the community add-on.
3. Add the Z-Wave JS integration again and uncheck "Use the Z-Wave JS Supervisor add-on" and enter the correct address for the community add-on in the URL field in the next step. This will update the existing config entry and abort the flow as already configured.
4. Enable the integration again.

Author: I don't think it reinstalls that quickly but I don't run addons and don't really know. We tell people my steps all the time on discord and they seem to do it. Wouldn't it be easier to have them reconfigure to uncheck the box then to disable it?

Can they still get into the addon to note their key when it's stopped?

Reviewer 2: Yes, they can check the config as long as the add-on isn't uninstalled.

Reviewer 2: > Wouldn't it be easier to have them reconfigure to uncheck the box then to disable it?

What do you mean?

Author: I mean is it better for them to disable it, or to reconfigure it and uncheck the box at step 1.

Reviewer 2: Currently just reconfiguring and unchecking the box to use the official add-on won't disconnect from the server and stop the add-on as I wrote above. I'll fix this, but since this docs update is targeting current branch, we should write what works right now.

Author: Ok, this should be fixed now.

{% endraw %}

#### Can I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

You can, but you cannot run them both at the same time. Only one of them can be active at the same time.

#### How do I switch between the Official Z-Wave JS add-on and Zwavejs2Mqtt?

Switching is easy and does not require renaming your devices.

1. Note your network security keys from your current add-on, then uninstall that add-on. **Do not remove the Z-Wave JS integration or you will lose all device and entity naming.**

2. Install and configure the new add-on, including setting those network security keys.

> 3. Re-add the Z-Wave integration to Home Assistant (_without_ removing it first). Add the the new/updated WebSocket URL. You may receive a popup stating that this Z-Wave network is already configured, but "under the hood" the WebSocket URL will be adjusted.

Reviewer 2: It's also possible to reconfigure the integration from the configuration panel `Re-Configure Server` button.

Author: I don't know, but I've been told that you can't change the websocket that way. Is that incorrect?

Switching does not require renaming your devices.

1. Disable the Z-Wave JS integration. **Do not remove the Z-Wave JS integration or you will lose all device and entity naming.** This will automatically stop the official Z-Wave JS add-on.

> 2. Note your network security keys from the official add-on, then uninstall that add-on.

Reviewer 2: We shouldn't uninstall the official add-on until we've updated the config entry by going through the config flow again.

5. The UI should confirm that the node was added and it will be immediately visible in Home Assistant. After a short while (seconds to minutes) the entities should also be created.
6. If the controller fails to add/find your device, cancel the inclusion process (to unblock your network again). In some cases it might help to first remove a node (exclusion) before you add it, even when the device has not been added to this Z-Wave network yet. Another approach would be to factory reset the device. Info about that is in the manual of your device.

If your device is included using S2 security, you may be prompted to enter a PIN number provided with your device. Often, this PIN is provided with the documentation _and_ is also printed on the device itself.

**Warning:**

1. **Do not move your Z-Wave stick to include devices.** This is no longer necessary and leads to broken routes. 
2. **Do not initiate device inclusion from the Z-Wave stick itself.** This is no longer supported.

#### Should I use `Secure Inclusion`?

> That depends. There are two generations of Z-Wave security, S0 and S2. 

Reviewer: ```suggestion
That depends. There are two generations of Z-Wave security, S0, and S2. 
```

S0 security imposes significant additional traffic on your mesh and is recommended only for devices that require security, such as door locks.

S2 security does not impose additional network traffic and provides additional benefits, such as detecting packet corruption. By default, Z-Wave JS attempts S2 security during inclusion if supported, falling back to S0 security only when necessary.

#### How do I remove (exclude) a device from my Z-Wave network?

1. In Home Assistant: open Configuration -> Integrations -> Z-Wave JS -> Configure.
2. Press `Remove node`.
3. Press `Start Exclusion`. The Z-Wave controller is now in exclusion mode and will not respond to other commands.
4. Put the device you want to remove in exclusion mode. Refer to its manual how this is done.
5. The UI should confirm that the node was removed and the device and entities will be removed from Home Assistant.

### Troubleshooting

#### I'm having a problem, what should I do first?

> _Many_ reported issues result from RF interference caused by the system's USB ports. This can manifest in many ways, including devices that won't include at all, devices that won't include securely, sensors with erroneous values (packets corrupted), delayed controlled of devices, or no ability to control devices. These problems can be intermittent, and they may be newly apparent after switching from `ozw`, though they existed before, as Z-Wave JS interviews devices and reports errors differently.

Reviewer: ```suggestion
_Many_ reported issues result from RF interference caused by the system's USB ports. This can manifest in many ways, including devices that won't include at all, devices that won't include securely, sensors with erroneous values (packets corrupted), delayed control of devices, or no ability to control devices. These problems can be intermittent, and they may be newly apparent after switching from `ozw`, though they existed before, as Z-Wave JS interviews devices and reports errors differently.
```

**All users are encouraged to use a USB extension cable to prevent such interference.** Please try such a cable before opening an issue or requesting support on Discord. It will nearly always be the first troubleshooting step that we ask you to take anyway.

After ensuring you are using an extension cable, perform a network heal.

The combination of these two steps corrects a large number of reported difficulties.

#### I have an Aeotec Gen5 controller, and it isn't detected on my Raspberry Pi 4?

> The first generation Gen5 controller has a known bug when plugged into a Pi 4 and possibly other systems. Aeotec released the Gen5+ stick to correct this bug. Gen5 users can plus their sticks into a USB 2.0 hub in order to overcome the issue.

Reviewer: ```suggestion
The first-generation Gen5 controller has a known bug when plugged into a Pi 4 and possibly other systems. Aeotec released the Gen5+ stick to correct this bug. Gen5 users can plus their sticks into a USB 2.0 hub in order to overcome the issue.
```

#### I do not see any entities created for my device in Home Assistant

> Entities will be created only after the node is ready (interview is completed). Also, note that some devices (like button remotes) do not create any entities but will only provide events when a button is pressed. See the events section on how to handle those events in your automations.

Reviewer: ```suggestion
Entities will be created only after the node is ready (the interview is completed). Also, note that some devices (like button remotes) do not create any entities but will only provide events when a button is pressed. See the events section on how to handle those events in your automations.
```

5. The UI should confirm that the node was removed and the device and entities will be removed from Home Assistant.

### Troubleshooting

#### I'm having a problem, what should I do first?

_Many_ reported issues result from RF interference caused by the system's USB ports. This can manifest in many ways, including devices that won't include at all, devices that won't include securely, sensors with erroneous values (packets corrupted), delayed control of devices, or no ability to control devices. These problems can be intermittent, and they may be newly apparent after switching from `ozw`, though they existed before, as Z-Wave JS interviews devices and reports errors differently.

**All users are encouraged to use a USB extension cable to prevent such interference.** Please try such a cable before opening an issue or requesting support on Discord. It will nearly always be the first troubleshooting step that we ask you to take anyway.

After ensuring you are using an extension cable, perform a network heal.

The combination of these two steps corrects a large number of reported difficulties.

#### I have an Aeotec Gen5 controller, and it isn't detected on my Raspberry Pi 4?

> The first-generation Gen5 controller has a known bug when plugged into a Pi 4 and possibly other systems. Aeotec released the Gen5+ stick to correct this bug. Gen5 users can plus their sticks into a USB 2.0 hub in order to overcome the issue.

Reviewer 2: ```suggestion
The first-generation Gen5 controller has a known bug when plugged into a Pi 4 and possibly other systems. Aeotec released the Gen5+ stick to correct this bug. Gen5 users can plug their sticks into a USB 2.0 hub in order to overcome the issue.
```

After ensuring you are using an extension cable, perform a network heal.

The combination of these two steps corrects a large number of reported difficulties.

#### I have an Aeotec Gen5 controller, and it isn't detected on my Raspberry Pi 4?

The first generation Gen5 controller has a known bug when plugged into a Pi 4 and possibly other systems. Aeotec released the Gen5+ stick to correct this bug. Gen5 users can plus their sticks into a USB 2.0 hub in order to overcome the issue.

#### I do not see any entities created for my device in Home Assistant

Entities will be created only after the node is ready (interview is completed). Also, note that some devices (like button remotes) do not create any entities but will only provide events when a button is pressed. See the events section on how to handle those events in your automations.

If you are certain that your device should have entities and you do not see them (even after a restart of Home Assistant Core), create an issue about your problem on the GitHub issue tracker.

#### I get a lot of sensor entities for my device that I'm probably never going to use

> The integration will add as many usable entities for you as possible from the information it retrieves from your Z-Wave devices. Entities that you don't want/like, can be disabled within the Home Assistant interface.

Reviewer 2: This is not specific to this integration. Let's remove this.
